build.gradle

```diff
@@ -29,13 +29,13 @@ repositories {
 }
 
 dependencies {
-	compile 'com.datastax.cassandra:cassandra-driver-core:2.1.7.1'
+	compile 'com.datastax.cassandra:cassandra-driver-core:3.0.1'
 	compile 'com.google.guava:guava:18.0'
 
 	testCompile 'org.spockframework:spock-core:1.0-groovy-2.4'
 	testCompile 'org.codehaus.groovy:groovy-all:2.4.4'
 	testRuntime 'io.netty:netty-all:4.1.0.Beta5'
-	testCompile 'org.cassandraunit:cassandra-unit:2.1.9.2'
+	testCompile 'org.cassandraunit:cassandra-unit:3.0.0.1'
```

Comment:
My gradle-fu is not too strong, but the whole problem happens to be in the test dependencies. 
However, I know that we can set the `sourceVersion` and `targetVersion` on the build. So I think we could create jars that are java7 compatible, but use java8 for the build process.

Is that something you would be comfortable with?